### DISCUSSION TITLE
Jedis 5.0.1 Cluster aware?

### DISCUSSION THREAD

**Discussion by sachingarg123** [xetorthio/jedis#3643] [General]
We are using Jedis 5.0.1 to connect with our Redis Cluster using Jedis Cluster.
We have given all Ips to Jedis Cluster. Wanted to understand few things

1. is client internally able to identify the hashslot of key and Node IP where it actually available in cluster environment.
2. Does client able to calculate hashslot internally?

**Comment by sazzad16** [comment]
1.a. Yes.
1.b. Yes, if it is able to successfully connect to at least one node.
2. Yes.

Note: After making the first successful contact, it will use the IPs replied by CLUSTER SLOTS command. So those IPs must be accessible.

**Reply by sachingarg123** [reply]
@sazzad16 , Thanks for the reply. CAn you also redirect me to code which has this logic. i couldn't find it in 5.0.1 version

**Reply by sazzad16** [reply]
@sachingarg123 There are four points mentioned in my answer. Which one's logic are you mostly looking for?

**Reply by sachingarg123** [reply]
@sazzad16 : I m looking for code where hashslot are selected while using keys and also code where nodes are selected

**Reply by sachingarg123** [reply]
@sazzad16 : is it ClusterConnectionProvider class in case of Cluster setup?

**Reply by sazzad16** [reply]
It's the combination of ClusterCommandExecutor, ClusterConnectionProvider and JedisClusterInfoCache.
